client/hosting/performance/hooks/usePerformanceReport.ts

```diff
@@ -0,0 +1,107 @@
+import { URL } from 'url';
+import { useState, useCallback, useEffect } from 'react';
+import { useUrlBasicMetricsQuery } from 'calypso/data/site-profiler/use-url-basic-metrics-query';
+import { useUrlPerformanceInsightsQuery } from 'calypso/data/site-profiler/use-url-performance-insights';
+import { TabType } from 'calypso/performance-profiler/components/header';
+
+export const usePerformanceReport = (
+	setIsSavingPerformanceReportUrl: ( isSaving: boolean ) => void,
+	refetchPages: () => void,
+	savePerformanceReportUrl: (
+		pageId: string,
+		wpcom_performance_report_url: { url: string; hash: string }
+	) => Promise< void >,
+	currentPageId: string,
+	wpcom_performance_report_url: { url: string; hash: string } | undefined,
+	activeTab: TabType
+) => {
+	const { url = '', hash = '' } = wpcom_performance_report_url || {};
+
+	const [ retestState, setRetestState ] = useState( 'idle' );
+
+	const {
+		data: basicMetrics,
+		isError: isBasicMetricsError,
+		isFetched: isBasicMetricsFetched,
+		isLoading: isLoadingBasicMetrics,
+		refetch: requeueAdvancedMetrics,
+	} = useUrlBasicMetricsQuery( url, hash, true );
+
+	const { final_url: finalUrl, token } = basicMetrics || {};
+
+	useEffect( () => {
+		if ( token && finalUrl && URL.canParse( finalUrl ) ) {
```

Comment:
Should we add this `URL.canParse` validation [here](https://github.com/Automattic/wp-calypso/blob/fix/performance-profiler-saving-bad-url/client/hosting/performance/hooks/useSitePerformancePageReports.ts#L149-L156) instead? So if someone calls this in the future from another place, we could reuse that validation.